src/Stripe.net/Services/_interfaces/IListable.cs

```diff
@@ -4,7 +4,7 @@ namespace Stripe
     using System.Threading.Tasks;
 
     public interface IListable<T, O>
-        where T : IStripeEntity
+        where T : IStripeEntity, IHasId
```

Comment:
We need the resource to have an ID in order to be able to paginate. I also added `IHasId` as a constraint on other interfaces where it makes sense to do so (i.e. everything except create).